tree-walk.c

```diff
@@ -752,6 +825,12 @@ enum follow_symlinks_result get_tree_entry_follow_symlinks(struct object_id *tre
 	return retval;
 }
 
+enum follow_symlinks_result get_tree_entry_follow_symlinks(struct object_id *tree_oid, const char *name, struct object_id *result, struct strbuf *result_path, unsigned *mode)
```

Comment:
There are only three callers of this function, currently. I think we can change the signature to accept that `gentle` flag.